src/platform/Darwin/PlatformManagerImpl.cpp

```diff
@@ -109,7 +109,12 @@ CHIP_ERROR PlatformManagerImpl::_StopEventLoopTask()
                 dispatch_suspend(mWorkQueue);
                 mIsWorkQueueSuspended         = true;
                 mIsWorkQueueSuspensionPending = false;
-                dispatch_semaphore_signal(mRunLoopSem);
+
+                // only dispatch if event loop not already stopped
```

Comment:
This has nothing to do with "not already stopped".  The real issue is that we are no longer calling `RunEventLoop`, so there is no semaphore, but we are shutting down the event loop from inside the event loop itself (by posting an event to it to call `StopEventLoopTask` instead of just calling `StopEventLoopTask` directly).

The way this stuff is generally meant to be used is:

1. When using RunEventLoop, call StopEventLoopTask from inside that event loop, to make RunEventLoop finish and unblock the caller. 
2. When using StartEventLoopTask, call StopEventLoopTask from whatever entity called StartEventLoopTask (and in particular, from outside the event loop itself).  This gives it the equivalent of join() semantics, where it guarantees that the event loop is cleanly shut down before StopEventLoopTask returns.

PlatformManager.h could be a bit clearer about documenting this...

Reply:
Updated code, I believe this change is not needed anymore.